src/server/game/Events/GameEventMgr.cpp

```diff
@@ -1071,50 +1072,81 @@ void GameEventMgr::LoadFromDB()
 void GameEventMgr::LoadHolidayDates()
 {
     uint32 oldMSTime = getMSTime();
+    uint32 dynamicCount = 0;
+    uint32 dbCount = 0;
 
-    WorldDatabasePreparedStatement* stmt = WorldDatabase.GetPreparedStatement(WORLD_SEL_GAME_EVENT_HOLIDAY_DATES);
-    PreparedQueryResult result = WorldDatabase.Query(stmt);
+    // Step 1: Generate dynamic holiday dates based on current year
+    time_t now = time(nullptr);
```

Comment:
```suggestion
    time_t const now = time(nullptr);
```

please make all other variables of your pr also `const` that can be `const`